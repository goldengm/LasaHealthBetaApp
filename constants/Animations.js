const Animations = {
  amieBranding: {
    // Logo: require("../assets/amie/amie_logo.png"),
    // LogoText: require("../assets/amie/amie_logo_text.png"),
    // iOSLogo: require("../assets/amie/amie_logo_text.png"),
    // androidLogo: require("../assets/amie/amie_logo_text.png"),
    // androidLogoText: require("../assets/amie/amie_logo_text.png"),
    // LogoTrans: require("../assets/amie/amie_logo_trans.png"),
    // LogoWhite: require("../assets/amie/amie_logo_white.png"),
  },
  garden: {
    flower: {
      // associado1: require("../assets/garden/flowers/animated/12162-associado.json"),
      rose1: require("../assets/garden/flowers/animated/17302-rose.json"),
      treeplant1: require("../assets/garden/flowers/animated/19087-treeplant.json"),
      windPlant1: require("../assets/garden/flowers/animated/23022-wind-plant.json"),
      plants1: require("../assets/garden/flowers/animated/24521-plants.json"),
      floralButterfly1: require("../assets/garden/flowers/animated/25797-concept-floral-branch-butterfly-and-tree.json"),
      flower1: require("../assets/garden/flowers/animated/29583-flower.json"),
      plantaLaranja1: require("../assets/garden/flowers/animated/33298-planta-laranja.json"),
      flower2: require("../assets/garden/flowers/animated/34496-plants.json"),
      greenFlower1: require("../assets/garden/flowers/animated/36932-green-flower.json"),
      // animatedPlantInPot1: require("../assets/garden/flowers/animated/41549-animated-plant-in-a-pot.json"),
      plantAnimation1: require("../assets/garden/flowers/animated/44201-plant-animation.json"),
      plant123: require("../assets/garden/flowers/animated/64993-plant-123.json"),
      plantMoving1: require("../assets/garden/flowers/animated/68866-plant-moving.json"),
      orangePlant1: require("../assets/garden/flowers/animated/70942-orange-plant.json"),
      // plant2: require("../assets/garden/flowers/animated/77528-plant.json"),
      wheatTwigs1: require("../assets/garden/flowers/animated/78957-wheat-twigs.json"),
      flowerPlantMoving1: require("../assets/garden/flowers/animated/92829-flower-plant-moving.json"),
      floweringPlant1: require("../assets/garden/flowers/animated/92886-flowering-plant.json"),
      flowerTreeplant: require("../assets/garden/flowers/animated/93007-flower-treeplant-animation.json"),
      // qrCode1: require("../assets/garden/flowers/animated/95381-qrcode.json"),
      // plant3: require("../assets/garden/flowers/animated/97679-plant.json"),
      houseplant1: require("../assets/garden/flowers/animated/99626-houseplant.json"),
      flower3: require("../assets/garden/flowers/animated/99690-flowers.json"),
      blueFlower1: require("../assets/garden/flowers/animated/99748-blue-flower.json"),

      aloevera1: require("../assets/garden/newFlowers/animated/aloevera-plant.json"),
      bamboo1: require("../assets/garden/newFlowers/animated/bamboo.json"),
      blossonTree1: require("../assets/garden/newFlowers/animated/blossom-tree.json"),
      borageFlower1: require("../assets/garden/newFlowers/animated/borage-flower.json"),
      cactus1: require("../assets/garden/newFlowers/animated/cactus.json"),
      cactus2: require("../assets/garden/newFlowers/animated/cactus_2.json"),
      chamomileFlower1: require("../assets/garden/newFlowers/animated/chamomile-flower.json"),
      cloverFlower1: require("../assets/garden/newFlowers/animated/clover-flower.json"),
      cornPlant1: require("../assets/garden/newFlowers/animated/corn-plant-and-pot.json"),
      flower4: require("../assets/garden/newFlowers/animated/flower_0.json"),
      flower5: require("../assets/garden/newFlowers/animated/flower_1.json"),
      flower6: require("../assets/garden/newFlowers/animated/flower_2.json"),
      flowerPot1: require("../assets/garden/newFlowers/animated/flower-pot.json"),
      flowerVase1: require("../assets/garden/newFlowers/animated/flower-vase.json"),
      gardening1: require("../assets/garden/newFlowers/animated/gardening.json"),
      grass1: require("../assets/garden/newFlowers/animated/grass.json"),
      lotusFlower1: require("../assets/garden/newFlowers/animated/lotus-flower.json"),
      pansyFlower1: require("../assets/garden/newFlowers/animated/pansy-flower.json"),
      pinkFlower1: require("../assets/garden/newFlowers/animated/pink-flower.json"),
      plantPot2: require("../assets/garden/newFlowers/animated/plant-pot_2.json"),
      roseFlower1: require("../assets/garden/newFlowers/animated/rose-flower.json"),
      seedlingFlowers1: require("../assets/garden/newFlowers/animated/seedling-flowers.json"),
      spaFlower1: require("../assets/garden/newFlowers/animated/spa-flower.json"),
      sunflower1: require("../assets/garden/newFlowers/animated/sunflower.json"),
    },
    foliage: {
      christmasTree1: require("../assets/garden/foliage/animated/christmas-tree-3891599.json"),
      christmasTree2: require("../assets/garden/foliage/animated/christmas-tree-3891601.json"),
      plant1: require("../assets/garden/foliage/animated/plant-4383521.json"),
      tree1: require("../assets/garden/foliage/animated/plant-a-tree-4386286.json"),
      treeGrow1: require("../assets/garden/foliage/animated/plant-growth-4307819.json"),
      tree2: require("../assets/garden/foliage/animated/tree-4383518.json"),
    },
    effect: {
      happyFlowerSparkle1: require("../assets/garden/gardenBackground/happyFlowSparkle1.json"),
      happyFlowerSparkle2: require("../assets/garden/gardenBackground/happyFlowSparkle2.json"),
    },
    test: {
      testTMP: require("../assets/garden/flowers/animated/test22.json"),
    },
  },
  ui: {
    doNotTouch1: require("../assets/ui/tutorial/handGestures/do-not-touch-2915065.json"),
    doubleClick1: require("../assets/ui/tutorial/handGestures/double-click-2915064.json"),
    doubleTap1: require("../assets/ui/tutorial/handGestures/double-tap-2915063.json"),
    dragAndDrop1: require("../assets/ui/tutorial/handGestures/drag-and-drop-2915062.json"),
    dragItDown1: require("../assets/ui/tutorial/handGestures/drag-it-down-2915061.json"),
    dragUp1: require("../assets/ui/tutorial/handGestures/drag-up-2915060.json"),
    fingerRotate1: require("../assets/ui/tutorial/handGestures/finger-rotate-2915058.json"),
    fingerTouchGestures1: require("../assets/ui/tutorial/handGestures/finger-touch-gestures-2915066.json"),
    rotateGesture1: require("../assets/ui/tutorial/handGestures/rotate-gesture-2915057.json"),
    rotateLeft1: require("../assets/ui/tutorial/handGestures/rotate-left-2915056.json"),
    rotateRightGesture1: require("../assets/ui/tutorial/handGestures/rotate-right-gesture-2915055.json"),
    scrollDown1: require("../assets/ui/tutorial/handGestures/scroll-down-2915054.json"),
    scrollUp1: require("../assets/ui/tutorial/handGestures/scroll-up-2915053.json"),
    slideLeftAndRight1: require("../assets/ui/tutorial/handGestures/slide-left-and-right-2915052.json"),
    swipeRight1: require("../assets/ui/tutorial/handGestures/swipe-right-2915051.json"),
    tap1: require("../assets/ui/tutorial/handGestures/tap-2915059.json"),
  },
};

export function getAnimationFromPath(path) {
  const pathParts = path.split(".");
  if (!(pathParts == null) && pathParts.length == 4) {
    return JSON.parse(
      JSON.stringify(Animations[pathParts[1]][pathParts[2]][pathParts[3]])
    );
  } else {
    return null;
  }
}

export default Animations;
