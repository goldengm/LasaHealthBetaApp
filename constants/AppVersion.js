export const APP_VERSION = '1.1.6';
export const DB_VERSION = '0.6.4';
export const DEBUG_TESTING = true;
