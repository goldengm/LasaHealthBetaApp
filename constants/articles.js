export default [
  {
    title: "Migraine: What Your Doctor Doesn’t Know",
    image:
      "https://images.everydayhealth.com/images/what-your-doctor-may-not-know-about-migraines-1440x810.jpg?w=1110",
    cta: "View article",
    horizontal: true,
    url: "https://www.everydayhealth.com/migraine/what-your-doctor-doesnt-know/",
  },
  {
    title:
      "Hypertension Symptoms Often Mistaken for Menopause in Middle-Aged Women",
    image:
      "https://images.everydayhealth.com/images/cardiovascular-disease-often-overlooked-in-middle-aged-women-1440x810.jpg?w=1110",
    cta: "View article",
    url: "https://www.everydayhealth.com/womens-health/hypertension-symptoms-in-middle-aged-women-often-mistaken-as-menopause/",
  },
  {
    title:
      "Study Explores Promising New Approach for Treating Premenstrual Dysphoric Disorder (PMDD)",
    image:
      "https://images.everydayhealth.com/images/promising-treatment-for-premenstrual-dysphoric-disorder-pmdd-1440x810.jpg?w=1110",
    cta: "View article",
    url: "https://www.everydayhealth.com/womens-health/study-explores-promising-new-approach-for-treating-premenstrual-dysphoric-disorder/",
  },
  {
    title:
      "12 Factors to Consider When Choosing the Best Birth Control for You",
    image:
      "https://images.everydayhealth.com/images/cs-choosing-the-right-birth-control-for-you-722x406.jpg?w=1110",
    cta: "View article",
    url: "https://www.everydayhealth.com/womens-health/factors-to-consider-choosing-your-birth-control/",
  },
  {
    title: "10 Home Remedies to Relieve Menstrual Cramps",
    image:
      "https://images.everydayhealth.com/images/womens-health/10-ways-to-relieve-menstrual-cramps-rm-722x406.jpg?w=1110",
    cta: "View article",
    horizontal: true,
    url: "https://www.everydayhealth.com/treatment/womens-health/ways-to-relieve-period-cramps/",
  },
  {
    title:
      "Why You Shouldn’t Miss Your Ob-Gyn Wellness Visit Because of the Pandemic",
    image:
      "https://images.everydayhealth.com/images/coronavirus/tips-to-get-the-most-out-of-telemedicine-with-gynecologist-722x406.jpg?w=1110",
    cta: "View article",
    horizontal: true,
    url: "https://www.everydayhealth.com/womens-health/why-you-shouldnt-miss-your-ob-gyn-wellness-visit-because-of-the-pandemic/",
  },
];
