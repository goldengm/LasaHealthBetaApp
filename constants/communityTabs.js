export default communityTabs = {
  categories: [
    { id: "Endometriosis", title: "Endometriosis" },
    { id: "Uterine Fibroids", title: "Uterine Fibroids" },
    { id: "Polycystic Ovary Syndrome", title: "Polycystic Ovary Syndrome" },
    {
      id: "Sexually Transmitted Diseases",
      title: "Sexually Transmitted Diseases",
    },
    { id: "Gastroenterology", title: "Gastroenterology" },
    { id: "HIV", title: "HIV" },
    { id: "Ovarian and Cervical Cancer", title: "Ovarian and Cervical Cancer" },
    { id: "Autoimmune Diseases", title: "Autoimmune Diseases" },
    { id: "Depression and Anxiety", title: "Depression and Anxiety" },
    { id: "Pregnancy Issues", title: "Pregnancy Issues" },
    { id: "Breast Cancer", title: "Breast Cancer" },
    { id: "Heart Disease", title: "Heart Disease" },
    { id: "Osteoporosis", title: "Osteoporosis" },
    { id: "Chronic Pain", title: "Chronic Pain" },
    { id: "Lupus", title: "Lupus" },
    { id: "Multiple Sclerosis", title: "Multiple Sclerosis" },
    { id: "Migraine Headaches", title: "Migraine Headaches" },
    { id: "Painful Periods", title: "Painful Periods" },
    { id: "Insomnia", title: "Insomnia" },
  ],
};
