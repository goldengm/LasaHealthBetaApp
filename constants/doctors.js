export default [
  {
    title: "Amanda M. Cleveland, MD",
    image:
      "https://intermountainhealthcare.org/-/media/images/modules/physician-directory/profiles/c/cleveland-amanda-md-2.jpg?mh=320&mw=320",
    cta: "View profile",
    // horizontal: true,
    rating: 4.6,
    url: "https://intermountainhealthcare.org/find-a-doctor/c/cleveland-amanda-m/",
  },
  {
    title: "Brooke L. Hansen, MD",
    image:
      "https://intermountainhealthcare.org/-/media/images/modules/physician-directory/profiles/h/hansen-brooke-img.jpg?mh=320&mw=320",
    cta: "View profile",
    rating: 4.7,
    url: "https://intermountainhealthcare.org/find-a-doctor/h/hansen-brooke-l/",
  },
  {
    title: "Brittany R. McColgan, MD",
    image:
      "https://intermountainhealthcare.org/-/media/images/modules/physician-directory/profiles/m/mccolgan-brittany-3.jpg?mh=320&mw=320",
    cta: "View profile",
    rating: 4.3,
    url: "https://intermountainhealthcare.org/find-a-doctor/m/mccolgan-brittany-r/",
  },
  {
    title: "Michael L. Hadley, MD",
    image:
      "https://intermountainhealthcare.org/-/media/images/modules/physician-directory/profiles/h/hadley-michael.jpg?mh=320&mw=320",
    cta: "View profile",
    rating: 4.5,
    url: "https://intermountainhealthcare.org/find-a-doctor/h/hadley-michael-l/",
  },
  {
    title: "Chelsea S. Mutscheller, MD",
    image:
      "https://intermountainhealthcare.org/-/media/images/modules/physician-directory/profiles/m/mutscheller-chelsea.jpg?mh=320&mw=320",
    cta: "View profile",
    // horizontal: true,
    rating: 4.8,
    url: "https://intermountainhealthcare.org/find-a-doctor/m/mutscheller-chelsea-s/",
  },
  {
    title: "David A. Vande Merwe, MD",
    image:
      "https://intermountainhealthcare.org/-/media/images/modules/physician-directory/profiles/v/vande-merwe-david.jpg?mh=320&mw=320",
    cta: "View profile",
    // horizontal: true,
    rating: 4.6,
    url: "https://intermountainhealthcare.org/find-a-doctor/v/vande-merwe-david-a/",
  },
  {
    title: "Taylor Permenter, DPT",
    image:
      "https://intermountainhealthcare.org/-/media/images/modules/physician-directory/profiles/p/taylor-permenter-dpt.jpg?mh=320&mw=320",
    cta: "View profile",
    // horizontal: true,
    rating: 4.8,
    url: "https://intermountainhealthcare.org/find-a-doctor/p/permenter-taylor/",
  },
];
