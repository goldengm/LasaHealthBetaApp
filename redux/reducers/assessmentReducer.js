import * as actions from '../constants/index';
import produce from 'immer';
//  FIXMETUNA:
//  FIXMETUNA:  This Chatbot is going to take a lot of thought and development
//  FIXMETUNA:
//  FIXMETUNA:  chat log contains the entire history of chats between the user and the chatbot
//  currentChatMessages only contains the current conversation and is moved to chatlog and erased
//  each time a conversation ends.
//  new conversation flag is used to allow different applications to flag that the AmieChatbot wants to
//  start a new conversation with the user (could be triggered by a user's sad input, etc)
//  there it's up to each page to watch for changes in the new conversation flag and bring up the
//  AmieChatbot interface which will display the new conversation
initialState = {
  assessmentResults: [],
};
export default asessmentReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case actions.SET_ASSESSMENT_RESULTS:
        draft.assessmentResults = [...draft.assessmentResults, ...action.payload];
        break;
      case actions.RESET_ASSESSMENT_RESULTS:
        draft.assessmentResults = [];
        break;

      default:
        break;
    }
});
