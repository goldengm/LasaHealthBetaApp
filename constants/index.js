import theme from "./Theme";
import articles from "./articles";
import Images from "./Images";
import Animations from "./Animations";
import tabs from "./tabs";
import styles from "./Style";
import communityTabs from "./communityTabs";
import utils from "./utils";

export {
  articles,
  theme,
  Images,
  tabs,
  communityTabs,
  styles,
  Animations,
  utils,
};
