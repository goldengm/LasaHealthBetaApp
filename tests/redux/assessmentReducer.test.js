import React from 'react';
import assessmentReducer from '../../redux/reducers/assessmentReducer';
import * as actions from '../../redux/constants/index';

const initialState = {
    assessmentResults: []
};

describe('Assessment general tests', () => {
  it('Testing SET_AND_DELETE Data For Assessment', () => {
    const myactions = [
      { type: actions.SET_ASSESSMENT_RESULTS, payload: [[1], [2], [0,1], [3]] },
      { type: actions.RESET_ASSESSMENT_RESULTS }
    ]

    const state = myactions.reduce((currentState, action) => assessmentReducer(currentState, action), initialState);
    expect(state).toMatchSnapshot();
  });

});
