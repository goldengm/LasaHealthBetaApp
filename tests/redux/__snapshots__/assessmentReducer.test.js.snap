// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assessment general tests Testing SET_AND_DELETE Data For Assessment 1`] = `
{
  "assessmentResults": [],
}
`;
